fuel-vm/src/tests/blockchain.rs

```diff
@@ -334,18 +500,20 @@ fn load_external_contract_code() {
         ]);
     }
 
+    // when
     load_contract.extend([
-        op::move_(reg_a, RegId::HP),  // r[a] := $hp
-        op::xor(reg_b, reg_b, reg_b), // r[b] := 0
-        op::ori(reg_b, reg_b, 12),    /* r[b] += 12 (will be
-                                       * padded to 16) */
-        op::ldc(reg_a, RegId::ZERO, reg_b), // Load first two words from the contract
-        op::move_(reg_a, RegId::SSP),       // r[b] := $ssp
-        op::subi(reg_a, reg_a, 8 * 2),      // r[a] -= 16 (start of the loaded code)
-        op::xor(reg_b, reg_b, reg_b),       // r[b] := 0
-        op::addi(reg_b, reg_b, 16),         // r[b] += 16 (length of the loaded code)
-        op::logd(RegId::ZERO, RegId::ZERO, reg_a, reg_b), // Log digest of the loaded code
-        op::noop(),                         // Patched to the jump later
+        op::move_(reg_a, RegId::HP),   // r[a] := $hp
+        op::xor(reg_b, reg_b, reg_b),  // r[b] := 0
+        op::ori(reg_b, reg_b, offset), // r[b] += offset
+        op::xor(reg_c, reg_c, reg_c),  // r[b] := 0
+        op::ori(reg_c, reg_c, len),    // r[b] += len
+        op::ldc(reg_a, reg_b, reg_c),  // Load first two words from the contract
+        op::subi(reg_a, RegId::SSP, 16), /* r[a] := $ssp - 16 (start of the loaded
+                                        * code) */
+        op::movi(reg_c, 16), // r[b] = 16 (length of the loaded code)
+        op::logd(RegId::ZERO, RegId::ZERO, reg_a, reg_c), /* Log digest of the
```

Comment:
It looks like you're loading `len` bytes from the contract, given that `len` is stored in `reg_c`. In the tests, you pass `0` for `len`, but the following opcodes hardcode `16`. If I understand correctly, `ldc` still loads the whole contract regardless of `$rC`, but I think the `subi`, `movi`, and `logd` opcodes here become incorrect. Are they even necessary for these tests? Maybe we can remove them.

Reply:
It fails the `ldc__load_external_contract_code` test without it.

But you're completely right. I think it just needs to be parameterized for the `len`. It just so happens that the padded length for that test is 16 :clown_face: